### DISCUSSION TITLE
The 'existing models' section in the documentation is misleading

### DISCUSSION THREAD

**Discussion by bigzero61** [neuropoly/axondeepseg#829] [Bug reports, suggestions for code refactoring]
ADS almost completely satisfies my need in TEM analysis. But there are a little mistakes in the 'existing models' section in the documentation that may confuse beginner. 
```
To download these models, you must first have AxonDeepSeg installed. Afterwards, run:

download_model -m <model name> -t <model type>
where <model name> is the full name (e.g. model_seg_rat_axon-myelin_SEM) and <model type> is either light or ensemble. To view available models and their details, run:

download_model --list
```

The command should be 'download_models'. And the the example for the <model name> is misleading. The full model name shown in the model list encounters key error in this command (e.g. 'unmyelinated-TEM' works instead of 'model_seg_unmyelinated_sickkids').

Besides, in the 'Using AxonDeepSeg' section, it should be noted that the 'ads_activate' command should be ran first to activate the environment before using 'axondeepseg' command.

Lastly, thank AxonDeepSeg and its contributors! This software did save a lot of my time!

**Comment by mathieuboudreau** [comment]
Thank you for raising this!

>Besides, in the 'Using AxonDeepSeg' section, it should be noted that the 'ads_activate' command should be ran first to activate the environment before using 'axondeepseg' command.

This shouldn't be the case I believe, right @joshuacwnewton ? On my mac at least, I don't need to activate to use any of the CLI commands.

>The command should be 'download_models'.

Good catch - I'll fix that ASAP.

>And the the example for the is misleading. The full model name shown in the model list encounters key error in this command (e.g. 'unmyelinated-TEM' works instead of 'model_seg_unmyelinated_sickkids').

Hmm good point - having duplicate names has always troubled me (one "for users" and one "for researchers" i.e. with more model details). Hum, this one I think we'll need to discuss internally on how the best way to solve it is correct, but I agree that here we ask for ` -m <model name> ` but when listing them,

```
2024-09-26 08:48:04.214 | INFO     | AxonDeepSeg.download_model:main:72 - dedicated-SEM
{'MODEL NAME': 'model_seg_rat_axon-myelin_SEM',
 'NUMBER OF CLASSES': 2,
 'OVERVIEW': 'Axon and myelin segmentation model trained on Scanning Electron '
             'Microscopy data.',
 'TRAINING DATA': '14.8 Mpx of SEM rat spinal cord images with a resolution of '
                  '0.13 um/px. The dataset used for this model is publicly '
                  'available here: '
                  'https://github.com/axondeepseg/data_axondeepseg_sem.'}
```

The one that explicitly is next to 'MODEL NAME'  isn't the one the CLI wants. @hermancollin just pinging you so that you're also aware.

**Reply by joshuacwnewton** [reply]
> This shouldn't be the case I believe, right @joshuacwnewton ? On my mac at least, I don't need to activate to use any of the CLI commands.

Yes, exactly! The `ads_activate` command shouldn't be necessary to use at all for most people.

Do you run into specific issues if you try to run `axondeepseg` without `ads_activate`, @bigzero61?

**Reply by bigzero61** [reply]
Yes.
Unfortunately, it would tell axondeepseg command not found without running ads_activate first.
My system is windows 11.

**Reply by mathieuboudreau** [reply]
Interesting - let me test if this also throws an error on our Github Actions running on windows!

**Reply by mathieuboudreau** [reply]
Here's the CI test run that just got initiated, https://github.com/axondeepseg/axondeepseg/actions/runs/11053908078/job/30709417729

I added these lines,

https://github.com/axondeepseg/axondeepseg/commit/eeeb361d4ad7d5392b62d9098d9337a866f51e3c

**Reply by mathieuboudreau** [reply]
@joshuacwnewton yep - this user issue was reproduced on our CI!

<img width="638" alt="Screenshot 2024-09-26 at 10 09 51 AM" src="https://github.com/user-attachments/assets/30c90074-bfbc-44bd-b911-892690c83ea4">

I guess we should probably also add a step that tests this (the CLIs without activating) in all our actions instances

**Reply by mathieuboudreau** [reply]
Ahh, maybe this is related to this warning/information given at the end of the installation for Windows here? 

https://github.com/axondeepseg/axondeepseg/blob/35d783c8edbb75f19f6c2ae33b2a30043252b4fa/install_ads.bat#L206-L225

@bigzero61 did you do these steps and still got the error?

**Reply by mathieuboudreau** [reply]
If this was the source of the issue, we should probably add a note in the documentation reminding Windows users of this additional step in the CLI section

**Reply by bigzero61** [reply]
Hi, @mathieuboudreau , I did all these steps. And I can actually call ads_napari directly without activating the environment. Besides, 'download_models' did not work as well until activating the ads environment. 

I also tried ADS on my Ubuntu22.04. The above issue also occurs on this system.

It is worth mentioning that ADS on Windows didn't work as properly as on Linux. For example, the unmyelinated mode of axondeepseg_morphometrics did not work on windows and showed error:
`024-09-26 17:50:40.927 | ERROR    | AxonDeepSeg.morphometrics.launch_morphometrics_computation:load_mask:104 - ERROR: Segmented unmyelinated axon mask for image: `H:\TEM\F2320-0001_c.png` is not present in the image folder. Please check that the unmyelinated axon mask is  located in the image folder. If it is not present, perform segmentation of the image first using ADS.`
Of note, I have renamed the unmyelinated masks produced by unmyelinated model from '*_axon.png' to '*_uaxon.png'. It works on Linux, but not on Windows.

Besides, the unmyelinated mask is spontaneously split into two masks, axon and myelin in Napari. Maybe it would be better there is only one unmyelinated mask and the mask be renamed as uaxon.png suffix automatically. 

It also seems that it is much easier to call GPU on the segmentation step on Linux. Though I have installed CUDA of proper version on Windows, it always showed me GPU was not found. And the speed seems to be over 10 times higher by using GPU.

**Comment by hermancollin** [comment]
@bigzero61 thank you for your feedback.

> Lastly, thank AxonDeepSeg and its contributors! This software did save a lot of my time!

Would you mind sharing what you are using AxonDeepSeg for? 

--- 

> The one that explicitly is next to 'MODEL NAME' isn't the one the CLI wants. @hermancollin just pinging you so that you're also aware.

@mathieuboudreau yes this would 100% warrant a small refactoring in model_cards.py - I just looked at the code and I think the MODEL_NAME fields should be the actual keys of the dict, to avoid any confusion.

**Reply by bigzero61** [reply]
Hi @hermancollin ,
I used AxonDeepSeq to measure the count of myelinated /unmyelinated axons and g-ratio in the corpus callosum in the  ischemic brain. Images were  captured by TEM.

It works perfectly on the uninjured brain to calculate g-ratio but the count of unmyelinated axons is still not proper enough (sadly, I also encounters some errors in the segmentation step). Don't know whether it could work in the injured brain as perfectly as in the healthy brain.

Thank again. This software would be definitely recommended to my colleagues and other relative  labs.
